onnxruntime/contrib_ops/cpu/triu.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include "core/platform/threadpool.h"
+#include "core/util/math_cpuonly.h"
+#include "Eigen/src/Core/Map.h"
+#include "Eigen/LU"
+#include "triu.h"
+#include <functional>
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_TYPED_KERNEL_EX(
+    Triu,
+    kMSDomain,
+    1,
+    float,
+    kCpuExecutionProvider,
+    KernelDefBuilder()
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    Triu);
+
+ONNX_OPERATOR_TYPED_KERNEL_EX(
```

Comment:
Because you are using MLTypeCallDispatcher in your kernel, you can just have a single ONNX_CPU_OPERATOR_KERNEL with a TypeConstaint that has float and double. See providers\cpu\math\sign.cc for an example.